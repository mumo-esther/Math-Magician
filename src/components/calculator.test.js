import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from './calculator';

test('renders correctly', () => {
  const { asFragment } = render(<Calculator />);
  expect(asFragment()).toMatchSnapshot();
});

test('handles multiple decimal points', () => {
  const { getByText } = render(<Calculator />);
  fireEvent.click(getByText('1'));
  fireEvent.click(getByText('.'));
  fireEvent.click(getByText('.'));
  expect(getByText('1.')).toBeInTheDocument();
});

test('handles consecutive operations', () => {
  const { getByText } = render(<Calculator />);
  fireEvent.click(getByText('1'));
  fireEvent.click(getByText('+'));
  fireEvent.click(getByText('-'));
  expect(getByText('-')).toBeInTheDocument();
});
