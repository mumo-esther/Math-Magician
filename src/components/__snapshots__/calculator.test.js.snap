// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="calculator-grid"
  >
    <div
      class="output"
    >
      <div
        class="previous-operand"
      />
      <div
        class="current-operand"
      >
        0
      </div>
    </div>
    <button
      type="button"
    >
      AC
    </button>
    <button
      type="button"
    >
      +/-
    </button>
    <button
      type="button"
    >
      %
    </button>
    <button
      class="colored"
      type="button"
    >
      รท
    </button>
    <button
      type="button"
    >
      7
    </button>
    <button
      type="button"
    >
      8
    </button>
    <button
      type="button"
    >
      9
    </button>
    <button
      class="colored"
      type="button"
    >
      x
    </button>
    <button
      type="button"
    >
      4
    </button>
    <button
      type="button"
    >
      5
    </button>
    <button
      type="button"
    >
      6
    </button>
    <button
      class="colored"
      type="button"
    >
      -
    </button>
    <button
      type="button"
    >
      1
    </button>
    <button
      type="button"
    >
      2
    </button>
    <button
      type="button"
    >
      3
    </button>
    <button
      class="colored"
      type="button"
    >
      +
    </button>
    <button
      class="span-two"
      type="button"
    >
      0
    </button>
    <button
      type="button"
    >
      .
    </button>
    <button
      class="colored"
      type="button"
    >
      =
    </button>
  </div>
</DocumentFragment>
`;
